library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -98,15 +102,19 @@
  */
 @org.apache.hadoop.classification.InterfaceStability.Evolving
 public abstract class AbstractFileInputOperator<T> implements InputOperator, Partitioner<AbstractFileInputOperator<T>>, StatsListener,
-    Operator.CheckpointListener, Operator.CheckpointNotificationListener
+    Operator.CheckpointListener, Operator.CheckpointNotificationListener, Operator.ActivationListener<OperatorContext>
 {
   private static final Logger LOG = LoggerFactory.getLogger(AbstractFileInputOperator.class);
 
+  protected static int MAX_FILE_PER_WINDOW = 1024;
```

Comment:
I followed the same design settings according to AbstractJdbcPollInputOperator. But yes I agree in this case maybe we should implement it as configurable so that users can adjust it according to their file size. I'll make the changes.